service.upnext/resources/lib/player.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+# GNU General Public License v2.0 (see COPYING or https://www.gnu.org/licenses/gpl-2.0.txt)
+
+from __future__ import absolute_import, division, unicode_literals
+from xbmc import getCondVisibility, Player, sleep
+from api import Api
+from state import State
+
+
+class UpNextPlayer(Player):
+    """Service class for playback monitoring"""
+    last_file = None
+    track = False
+
+    def __init__(self):
+        self.api = Api()
+        self.state = State()
+        Player.__init__(self)
+
+    def set_last_file(self, filename):
+        self.state.last_file = filename
+
+    def get_last_file(self):
+        return self.state.last_file
+
+    def is_tracking(self):
+        return self.state.track
+
+    def disable_tracking(self):
+        self.state.track = False
+
+    def onPlayBackStarted(self):  # pylint: disable=invalid-name
+        """Will be called when kodi starts playing a file"""
+        sleep(5000)  # Delay for slower devices, should really use onAVStarted for Leia
```

Comment:
can you use waitForAbort here? Sleep time is quite high and this may cause kodi to hang if requested for abort

Reply:
Checking, it's been that way for some time..as the saying goes if it's not broke don't fix it! But yeah will have a look